### PR TITLE
WIP: examples: notebooks: Moving between models

### PR DESCRIPTION
**Pull request by mHash1m**
Was waiting on #1092 (merged now)

- [x] Complete the notebook example

- [x] Test the notebook using test book

- [x] Add to docs with sphinx

### CODE REVIEW

File: docs/examples/notebooks/index.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Notebooks
+=========
+
+.. toctree::
+    :maxdepth: 2
+    :caption: Contents:
+
+    Moving_between_models
```
Comment (pdxjohnny) [inline docs/examples/notebooks/index.rst:8]: By convention we should have all lowercase docs

```suggestion
    moving_between_models
```

File: tests/notebooks/test_moving_between_models.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from testbook import testbook
+from dffml import IntegrationCLITestCase
+
+
+class TestNotebook(IntegrationCLITestCase):
+    @testbook("examples/notebooks/Moving_between_models.ipynb")
+    def test_stdout(tb, self):
+        tb.execute_cell(range(0, 5))
```
Comment (mHash1m) [inline tests/notebooks/test_moving_between_models.py:8]: So if you notice, I have removed the call to execute the whole notebook that I previously showed you on meet. Here, we initially execute till the 5th cell, excluding the call to `cached_download` in the notebook.

File: tests/notebooks/test_moving_between_models.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        tb.inject(
+            """
+            data_path = await cached_download(
+                "https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv",
+                "tests/notebooks/data/wine_quality.csv",
```
Comment (mHash1m) [inline tests/notebooks/test_moving_between_models.py:13]: As you suggested previously to mock the call to `cached_download` to give it a different path and gitignore that directory we download the example data to. Inject lets you to add some code to a notebook and execute it as part of the notebook, so I ignored the cell execution of the actual `cached_download` call and instead injected and executed this mock call.

File: tests/notebooks/test_moving_between_models.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                "789e98688f9ff18d4bae35afb71b006116ec9c529c1b21563fdaf5e785aea8b3937a55a4919c91ca2b0acb671300072c",
+            )
+            """
+        )
+        tb.inject(
+            "assert 'tests/notebooks/data/wine_quality.csv' in str(data_path)"
+        )
+        tb.execute_cell(range(7, 21))
```
Comment (mHash1m) [inline tests/notebooks/test_moving_between_models.py:21]: After that we just execute the rest of the cells and make our assertions

File: tests/notebooks/test_moving_between_models.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+from testbook import testbook
+from dffml import IntegrationCLITestCase
+
+
+class TestNotebook(IntegrationCLITestCase):
```
Comment (pdxjohnny) [inline tests/notebooks/test_moving_between_models.py:5]: Let's rebase in master which got rid of `IntegrationCLITestCase`

```suggestion
class TestNotebook(AsyncTestCase):
```

Comment (mHash1m) [reply tests/notebooks/test_moving_between_models.py]: @pdxjohnny couldn't get testbook to work with Async. Used unnittest.TestCase.

**Review by pdxjohnny — CHANGES_REQUESTED** [review]
Looking great! Nice work. Let's just make sure that we document the need for `pandoc` in `docs/contributing/docs.rst`

**Comment by pdxjohnny** [comment]
@mHash1m That would be great!! No I don't think your PR did that

**Pull request merged**